### DISCUSSION TITLE
MLFlow pipelines with Databricks workflows

### DISCUSSION THREAD

**Discussion by miwojc** [mlflow/mlflow#6231] [Q&A]
How to integrate MLFlow Pipelines with Databricks Workflows, or any other workflow orchestration tool?

**Comment by jinzhang21** [comment]
Good question, @miwojc ! We have not built in native support for Databricks Workflows with MLflow Pipelines yet. However it is a high priority topic on our roadmap discussion. For now, you can create a small notebook to schedule the pipeline as a Job on Databricks Workflows. For example, something like `%sh mlflow pipelines run`.

**Comment by mengxr** [comment]
@miwojc Do you mind sharing why you need that integration? Currently users can run MLflow Pipelines as a single step in workflow orchestration tools including Databricks Workflows.

**Reply by miwojc** [reply]
Just curious what's the recommended way. I like to think that steps in a workflow are data ingestion, preprocessing, feature engineering, train, test, etc. Maybe not necessarily all those stages in one step. So for example you can rerun failed steps, not wasting resources having to run entire end to end pipeline...

**Reply by mengxr** [reply]
MLP itself already understands what changed made by users and hence what steps need to be executed. So if you update estimator and run "evaluate" step, it will only execute the train step (reusing output from previous transform step) and then the evaluate step.

**Reply by miwojc** [reply]
Thank you. That's nice. Is MLP a kind of alternative solution for Workflows? Thinking where to position it in the existing processes / tools...

**Reply by mengxr** [reply]
MLP provides ML pipeline abstractions. To efficiently execute the pipeline, we need a cache-aware pipeline execution engine. We don't want to build a new engine (using `make` now).  It might be beter to adopt an existing one or support compiling down to several.

**Comment by ahdbilal** [comment]
@miwojc if you did get the chance to use the experimental version of the pipeline, we would love to talk to you to get your feedback. if interested, you can use this [form](https://docs.google.com/forms/d/e/1FAIpQLSeYh6-u0qFn7AYv06Kiph09oZq2PW7Vx3Z_qVjFLE5mfMWQFA/viewform?usp=sf_link) to sign up and we will reach out to you via email. Thank you!
